encrypted-config-value/src/main/java/com/palantir/config/crypto/EncryptedValue.java

```diff
@@ -18,21 +18,32 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 
+import com.google.common.io.BaseEncoding;
 import com.palantir.config.crypto.algorithm.Algorithm;
 import com.palantir.config.crypto.algorithm.Algorithms;
 import java.io.IOException;
+import javax.annotation.Nullable;
 import org.immutables.value.Value;
 
 @Value.Immutable
 public abstract class EncryptedValue {
 
     private static final String PREFIX = "enc:";
 
-    public abstract String encryptedValue();
+    @Nullable
+    public abstract String getAlgorithm();
```

Comment:
I picked "AES" since it was what we were tagging the keys with - "AES-256-GCM" would be a full description if you'd prefer to switch to that. I don't think it's necessary to explicitly state the IV size since there is a single canonical value of 92 bits. The IV size cannot be changed at all in most AES cipher modes. Even though it can with GCM, it's not a thing that happens very commonly (and is actively discouraged as #55 notes).